Sources/Plasma/FeatureLib/pfConsole/pfConsole.cpp

```diff
@@ -1002,6 +953,120 @@ void    pfConsole::IUpdateTooltip()
     }
 }
 
+//// IExecuteWorkingLine /////////////////////////////////////////////////////
+
+void    pfConsole::IExecuteWorkingLine()
+{
+    int   i, eol;
+    char* c;
+
+    // leave leading space for Python multi lines (need the indents!)
+    if (fPythonMultiLines == 0) {
+        // Clean up working line by removing any leading whitespace
+        for (c = fWorkingLine; *c == ' ' || *c == '\t'; c++);
+        for (i = 0; *c != 0; i++, c++)
+            fWorkingLine[i] = *c;
+        fWorkingLine[i] = 0;
+        eol = i;
+    }
+
+    if (fWorkingLine[0] == 0 && !fHelpMode && !fPythonMode) {
+        // Blank line--just print a blank line to the console and skip
+        IAddLine("");
+        return;
+    }
+
+    // only save history line if there is something there
+    if (fWorkingLine[0] != 0) {
+        // Save to history
+        strcpy(fHistory[fPythonMode].fData[fHistory[fPythonMode].fCursor], fWorkingLine);
+        fHistory[fPythonMode].fCursor = (fHistory[fPythonMode].fCursor < kNumHistoryItems - 1) ? fHistory[fPythonMode].fCursor + 1 : 0;
+        fHistory[fPythonMode].fRecallCursor = fHistory[fPythonMode].fCursor;
+    }
+
+    // EXECUTE!!! (warning: DESTROYS fWorkingLine)
+    if (fHelpMode) {
+        if (fWorkingLine[0] == 0) {
+            IPrintSomeHelp();
+        } else if (stricmp(fWorkingLine, "commands") == 0) {
+            char empty[] = "";
+            fEngine->PrintCmdHelp(empty, IAddLineCallback);
```

Comment:
I get that this is just refactored code from elsewhere, but is there a reason NOT to get rid of this first line and just use `fEngine->PrintCmdHelp("", IAddLineCallback)`? It feels strange to create a char array for the empty string, but maybe there's a C/C++/pointer-y reason that I'm not thinking of. The `empty` variable is never used again...

Reply:
See above, I didn't want to go down the rabbit hole of refactoring code that I was just moving to a new location. That would greatly increase the scope and complexity of this changeset.